[Althea Chua] iP

chuababyy opened the pull request:
# DukePro
> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

 - [x] Managing tasks
 - [ ] Managing deadlines (coming soon)
 - [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public static void main(String[] args) {
        new Duke(filePath).run();
    }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

chuababyy committed:
Level 0: Rename, Greet, Exit

chuababyy committed:
Level-1 Echo

chuababyy committed:
Level-2 Add, List

chuababyy committed:
Level-2 Add, List

chuababyy committed:
Level-3 Mark as Done

chuababyy committed:
Level-4 ToDos, Events, Deadlines

chuababyy committed:
Level-5: Handle Errors

chuababyy committed:
Level-5 Handle Errors

chuababyy committed:
Level-6 Delete

chuababyy committed:
A-Enums

chuababyy committed:
Add new feature: Saving tasks to file

chuababyy committed:
Merge branch 'branch-Level-7'

Add new feature

chuababyy committed:
Change date format

chuababyy committed:
Merge branch 'branch-Level-8'

Change date format

chuababyy committed:
Add more classes

chuababyy committed:
Merge branch 'A-MoreOOP'

Restructure files for OOP concepts

chuababyy committed:
Move all files into duke package

chuababyy committed:
Merge branch 'add-gradle-support'
Merge branch 'add-gradle-support'

chuababyy committed:
Add JUnit test cases

chuababyy committed:
Add support to create Jar file

chuababyy committed:
Add new header comments

chuababyy committed:
Merge branch 'branch-A-JavaDoc'

Add new header comments

chuababyy committed:
Revert "Add new header comments"

This reverts commit adb07e7dc104a358dc5f50fadd21795bbb1d2186.

Did not add parallel branch

chuababyy committed:
Undo revert

chuababyy committed:
Add JavaDocs header comments

chuababyy committed:
Merge branch 'branch-A-JavaDoc'

chuababyy committed:
Change code to follow coding standards

chuababyy committed:
Merge branch 'branch-A-CodingStandard'

Change code to follow coding standard

chuababyy committed:
Add new feature to find tasks

chuababyy committed:
Merge branch 'branch-Level-9'

Add new feature: Find tasks which match keyword

chuababyy committed:
Add checkstyle files

chuababyy committed:
Merge branch 'branch-A-Checkstyle'

Add checkstyle files and change code to comply with checkstyle

butteredyakiimo commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,67 @@
+        ui.showWelcome();
+        boolean isNotBye = true;
+        while (isNotBye) {
+            isNotBye = parser.createTaskAction(scanner.nextLine());
```
I like how you check for the terminating condition here. The use of `isNotBye` makes it quite intuitive to understand when the program should exit.

butteredyakiimo commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,79 @@
+     * @param start Start date to be compared.
+     * @param end End date to be compared.
+     */
+    public boolean isStartDateBefore(LocalDateTime start, LocalDateTime end) {
```
I think a more intuitive name for this could be `isStartBeforeEnd`! What do you think?

butteredyakiimo commented on src/main/java/duke/FileStorage.java:
```diff
@@ -0,0 +1,151 @@
+        file.mkdirs();
+
+        if (file.exists()) {
+            boolean isDeleteFileSuccess = file.delete();
```
I think instead of deleting and creating a new file each time the tasklist updates, an alternative would be to to use `writer.write(task.toFileString())` instead! This method will overwrite the existing data, allowing you to use the same file for every update :D What do you think?

butteredyakiimo commented on src/main/java/duke/FileStorage.java:
```diff
@@ -0,0 +1,151 @@
+     * @param date The date string to be converted.
+     * @return A LocalDateTime object representing the date and time.
+     */
+    public LocalDateTime setDate(String date) {
```
I think a more intuitive name for this method could be `convertToDate`! what do you think?

butteredyakiimo commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,225 @@
+     * @param date The date string in the specified format.
+     * @return A LocalDateTime object representing the parsed date and time.
+     */
+    public LocalDateTime setDate(String date) {
```
similar to previous comment on the same method in storage class!

butteredyakiimo commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,225 @@
+     * @param input The user input to be processed.
+     * @return {@code true} if the application should continue running. {@code false} if it should exit.
+     */
+    public boolean createTaskAction(String input) {
```
I like the rationale behind this method! I think a more intuitive name for this method could be `isTaskCreated`, so that when false is returned, it means that the task was not created, and the application should stop running. What do you think?

butteredyakiimo commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,225 @@
+            ui.showEmptyMessage();
+            return true;
+
+        case UNKNOWN:
```
I think leaving the `UNKNOWN` case to the last case would make reading this block of code smoother!

butteredyakiimo reviewed (commented):
LGTM :+1: Clever use of switch statements and booleans in implementing code logic. Just some minor naming issues to fix!

chuababyy committed:
Add GUI for Duke

chuababyy committed:
Merge branch 'branch-Level-10'
Add GUI for Duke

xenosf commented on README.md:
It seems like when you moved your classes into packages, the class names ("Duke" here) got auto-reformatted to contain the package name. Since the "Duke" here is not a class name, should these be changed back? (`duke.Duke` -> `Duke`)

xenosf commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,84 @@
+    //     */
+    //    public static void main(String[] args) {
+    //        new Duke().run();
+    //    }
```
Should this unused code be deleted instead of commented out to keep the code cleaner?

xenosf commented on src/main/java/duke/components/Parser.java:
```diff
@@ -0,0 +1,227 @@
+            return CommandType.FIND;
+        } else {
+            return CommandType.UNKNOWN;
+        }
```
This if statement is very long -- would it be more readable if it was a switch-case instead?
You can consider using something like [`CommandType.valueOf()`](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/Enum.html#valueOf(java.lang.Class,java.lang.String)) to make the code more concise.

xenosf commented on src/main/java/duke/components/Parser.java:
```diff
@@ -0,0 +1,227 @@
+            } else {
+                return ui.showFind(list);
+            }
+
```
I like the use of line breaks to separate logical blocks and make the code easier to read!

xenosf commented on src/main/java/duke/components/TaskList.java:
Should the unused code blocks in this file be deleted instead of commented out to keep the code cleaner? If the old code is needed for reference, it is possible to view past versions using Git

xenosf commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,54 @@
+        this.deadline = deadline;
+    }
+
+    /**
```
This

```java
/**
 * {@inheritDoc}
 */
```

might not be necessary - If the entire comment is inherited from the overridden method, the whole JavaDoc is inherited as well.

chuababyy committed:
Add assertions

Task and Parser class has assumptions made as various points

There is a lack of validation for these assumptions

Java assert function is used to validate these assumptions

Assertions is one of the quickest and most effective ways to
detect and correct bugs

chuababyy committed:
Merge pull request #1 from chuababyy/branch-A-Assertions

Add assertions

chuababyy committed:
Change code for code quality

chuababyy committed:
Merge pull request #2 from chuababyy/branch-A-CodeQuality

Change code for code quality

chuababyy committed:
Add new feature: Mass ops

chuababyy committed:
Add new feature: Mass ops

chuababyy committed:
Merge branch 'branch-C-MassOps'
Add new feature: MassOps

chuababyy committed:
Change GUI

chuababyy committed:
Merge branch 'branch-A-BetterGui'

Change GUI

chuababyy committed:
Update README

chuababyy committed:
Merge branch 'branch-A-UserGuide'

Update README

chuababyy committed:
Fix bugs, Update Javadocs

chuababyy committed:
Change config
